ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -1,22 +1,211 @@
+import json
 import math
-import time
+import typing as t
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.llmobs._constants import RUNNER_IS_INTEGRATION_SPAN_TAG
+
+
+logger = get_logger(__name__)
+
+RAGAS_DEPENDENCIES_PRESENT = None
+
+try:
+    from ragas.llms import llm_factory
+    from ragas.llms.output_parser import RagasoutputParser
+    from ragas.llms.output_parser import get_json_format_instructions
+    from ragas.metrics import faithfulness
+    from ragas.metrics.base import ensembler
+    from ragas.metrics.base import get_segmenter
+
+    from .utils import StatementFaithfulnessAnswers
+    from .utils import StatementsAnswers
+
+    RAGAS_DEPENDENCIES_PRESENT = True
+except ImportError as e:
+    logger.warning("RagasFaithfulnessEvaluator is disabled because Ragas requirements are not installed", exc_info=e)
+    RAGAS_DEPENDENCIES_PRESENT = False
 
 
 class RagasFaithfulnessEvaluator:
     LABEL = "ragas_faithfulness"
     METRIC_TYPE = "score"
 
     def __init__(self, llmobs_service):
-        self.llmobs_service = llmobs_service
+        self.enabled = True
+        if not RAGAS_DEPENDENCIES_PRESENT:
+            self.enabled = False
+            return
+
+        self.llmobs = llmobs_service
+        self.faithfulness = faithfulness
+        if not self.faithfulness.llm:
+            self.faithfulness.llm = llm_factory()
+
+        self.statements_output_instructions = get_json_format_instructions(StatementsAnswers)
+        self.statements_output_parser = RagasoutputParser(pydantic_object=StatementsAnswers)
+        self.faithfulness_output_instructions = get_json_format_instructions(StatementFaithfulnessAnswers)
+        self.faithfulness_output_parser = RagasoutputParser(pydantic_object=StatementFaithfulnessAnswers)
+        self.sentence_segmenter = get_segmenter(language=self.faithfulness.nli_statements_message.language, clean=False)
 
     def evaluate(self, span):
```

Comment:
Let's add a brief docstring to describe what this evaluator is doing, and what it should return. I think we're relying too much on third party library details to understand this

Reply:
For sure, will add more descriptive variable names + docstrings!